src/webhook/upgrade_operator/fetch_upgradeconfig.py

```diff
@@ -1,19 +1,25 @@
 import os
 from kubernetes import client, config
 from kubernetes.client.rest import ApiException
+from kubernetes.config.config_exception import ConfigException
 
 # Use KUBECONFIG from pod else consume from local ~/.kube/config
```

Comment:
I'm still concerned that this is being done at the file scope. Side-effecty imports should be used very sparingly, and I feel it's not justified in this case.

Reply:
[Later] Actually, this is triggering your CI failure (although IMO this indicates a bug in the kubernetes client library, which should be catching that condition and raising `ConfigException` instead).

So yeah, I think you should wrap this setup in a method. If you want singleton behavior, track it in a global variable.

That would also make testing easier in general, because you could mock the client.

Reply:
I have added changes. Can you review from Python point of view? Have tried to incorporate the changes from what I understood from your suggestions :) Let me know if it's acceptable with you.